### PR TITLE
[C++] [Graph] task_prerequisite_harikrishnan.cpp

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
- [x] Code is added in right directory(Folder).
- [x] Brief Description of Solution is given.
- [x] Time and Space Complexity is mentioned
- [x] Code is well indented.
- [x] Comments are clearly written wherever necessary
- [x] Code is successfully compiled and passes all test cases.

### CODE REVIEW

**Comment by anjali1903** [comment]
Expected output for {0,1} is YES.

**Comment by Harikrishnan6336** [comment]
Hey @anjali1903, I have made the necessary changes.

**Comment by anjali1903** [comment]
It still gives the wrong answer.

**Comment by Harikrishnan6336** [comment]
Hey @anjali1903, You mean the input :

1
0 1 

and its  expected output is "YES" right?

**Comment by anjali1903** [comment]
Yes

**Comment by Harikrishnan6336** [comment]
Hey @anjali1903  The current code gives the desired output "YES".

File: 9. Graph/task_prerequisite/task_prerequisite_harikrishnan.cpp [hunk 1]
```diff
@@ -37,11 +37,16 @@ bool checkCycle(int n, vector<int> adj[], vector<bool>& vis, vector<bool>& rec)
     rec[n] = false;  // Remove the node from the recursive stack. 
     return false;   
 }
- 
+
 void solve()
 {
     int N,K,u,v;
     cin>>N>>K;                 // Input the string
+    if(N == 0)            // No tasks gievn.
```
Comment (anjali1903) [inline 9. Graph/task_prerequisite/task_prerequisite_harikrishnan.cpp:45]: You only added this snippet of code and this is not enough for the logic to be correct.

Comment (anjali1903) [reply 9. Graph/task_prerequisite/task_prerequisite_harikrishnan.cpp:45]: For the input
2 1
0 1
Expected output is YES

**Comment by Harikrishnan6336** [comment]
Hey @anjali1903. Could you Please review now.

**Comment by anjali1903** [comment]
Amazing!

**Pull request merged**